commit cf1a31a8ea464cf5e5189e0a5ee04792a8172ff3
Author: Mark Thomas <markt@apache.org>
Date:   Wed Apr 27 09:41:02 2016 +0000

    Review of Map / ConcurrentMap / ConcurrentHashMap
    attributes is non-final so it can be replaced. Define the field as ConcurrentMap to ensure that thread-safety is maintained if this happens.
    
    git-svn-id: https://svn.apache.org/repos/asf/tomcat/trunk@1741181 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/java/org/apache/catalina/session/StandardSession.java b/java/org/apache/catalina/session/StandardSession.java
index f590604..9434832 100644
--- a/java/org/apache/catalina/session/StandardSession.java
+++ b/java/org/apache/catalina/session/StandardSession.java
@@ -36,6 +36,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.atomic.AtomicInteger;
 
 import javax.servlet.ServletContext;
@@ -142,7 +143,7 @@ public class StandardSession implements HttpSession, Session, Serializable {
     /**
      * The collection of user data attributes associated with this Session.
      */
-    protected Map<String, Object> attributes = new ConcurrentHashMap<>();
+    protected ConcurrentMap<String, Object> attributes = new ConcurrentHashMap<>();
 
 
     /**
